src/styles/builders.js

```diff
@@ -385,45 +430,142 @@ function addFan (coord, nA, nC, nB, uA, uC, uB, signed, numTriangles, constants)
     // because we are going to add more triangles.
     indexPairs(constants);
 
+    // Initial parameters
     var normCurr = Vector.set(nA);
     var normPrev = [0,0];
+    // Calculate the angle between A and B 
+    var angle_delta = Vector.angleBetween(nA, nB);
+
+    // If the numbers of triangles is 3 use on rounds caps and joins
+    if (numTriangles === 3) {
+        // ... if that's the case try to simplify the number of triangles
+        var w = constants.halfWidth*2;
 
-    var angle_delta = Vector.dot(nA, nB);
-    if (angle_delta < -1) {
-        angle_delta = -1;
+        // Core ecuation taked from here (http://slabode.exofire.net/circle_draw.shtml) adapted from circle to what ever angle we have
+        var dist = Vector.length(Vector.sub( Vector.mult(nA,w), Vector.mult(nB,w)))/constants.units_per_pixel;
+        numTriangles = Math.ceil( (100*Math.sqrt(dist)/360)*angle_delta );
+        numTriangles = Math.min(15, Math.max(1,numTriangles));  // Limit the max and min
+
+        // Debug it
+        // console.log("Circle: dist:",dist,"angle:",angle_delta, "U/pixel:",constants.units_per_pixel," -> # triangles", numTriangles);
+
+        // If is a cap and have less than 2 triangles skip
+        if (angle_delta >= 3.14 && numTriangles < 2) {
```

Comment:
if is a join will probably assign more than 2 triangles for over 180 degrees. This scape call is just for caps that in low zoom levels have one or 0 triangles assigned... to avoid extra work